Bluetooth: Add missing store `net_buf_ref` return value

LingaoM opened the pull request:
From @jhedberg 's review of (https://github.com/zephyrproject-rtos/zephyr/pull/35702)

No code should ever call net_buf_ref() without storing the return value to a new pointer.

And also related to PR(https://github.com/zephyrproject-rtos/zephyr/pull/36259).

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

jhedberg commented on subsys/bluetooth/controller/hci/hci_driver.c:
```diff
@@ -519,8 +519,7 @@ static void recv_thread(void *p1, void *p2, void *p3)
 			 */
 			frag = buf;
-			net_buf_ref(frag);
-			buf = net_buf_frag_del(NULL, frag);
+			buf = net_buf_frag_del(NULL, net_buf_ref(frag));
```
Actually, I think the more correct thing would be to change `frag = buf;` into `frag = net_buf_unref(buf)`.

The reasoning: the code wants to store `frag` so that I can be later passed to `bt_recv()`. The `net_buf_frag_del()` line is just an iterator statement, and in fact the code should more correctly be passing `buf` as the input parameter (kind of like `i = i + 1;`) i.e. `buf = net_buf_frag_del(NULL, buf);`.

Also, please split the controller change into a separate commit. This is good practice in general, and it'll also give you a boost in contribution statistics 😉

cvinayak replied:
When you split into controller commit, do not add the `Fix` word in the commit message (i tend to relate `Fix` to associate with bug fix and needing backport if need be)

LingaoM committed:
Bluetooth: Add missing store `net_buf_ref` ret value

... related to(#36259)

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

LingaoM committed:
Bluetooth: Controller: Add `net_buf_ref` ret value to variables

Return value should assigns to variables instead of ignoring them.

Related to PR(#36259).

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

jhedberg commented on subsys/bluetooth/controller/hci/hci_driver.c:
```diff
@@ -518,8 +518,7 @@ static void recv_thread(void *p1, void *p2, void *p3)
-			frag = buf;
-			net_buf_ref(frag);
+			frag = net_buf_ref(buf);
 			buf = net_buf_frag_del(NULL, frag);
```
It's a small difference, but as I said in my previous comment, I think it's more correct to pass `buf` as the second parameter to frag_del(). The way that frag_del() is defined it's implied that you loose ownership of what you pass to it, and shouldn't use it afterwards, however the code does specifically use `frag` to pass it to `bt_recv()` whereas the original `buf` pointer is what's not needed anymore (since you assign the return value to it).
